### ISSUE TITLE
Status Bar Icons Disappeared

### ISSUE THREAD

**Issue by mchlp** [ShyykoSerhiy/vscode-spotify#42]
The controls that were in the status bar have disappeared. However, the keyboard shortcuts and controlling the extension through the command palette still works.

**Comment by ShyykoSerhiy** [comment]
That's odd. What version of OS, vscode do you use?

**Comment by mchlp** [comment]
OS: Ubuntu 18.04 LTS x86_64
VSCode Version: 1.25.1
I've tried reinstalling vscode and the extension and it didn't seem to work.

**Comment by ShyykoSerhiy** [comment]
It's gonna take me sometime to investigate. I need to bring VM up. Thx for report.

**Comment by ShyykoSerhiy** [comment]
This extension uses spotify's built in http web server on linux. Sometimes it crashes and fails to restart. Reboot and/or spotify reopen may fix this. If you want you can try this. I'm still going to start up the VM meanwhile.

**Comment by mchlp** [comment]
Rebooting now... The functionality of the extension still works (I can still use the keyboard shortcuts and commands through the commander palette to control Spotify). It's just that nothing from the extension is appearing in the status bar.

**Comment by ShyykoSerhiy** [comment]
It uses separate thing for next/previous song controls (dbus) so they can still work while command palette isn't.

**Comment by CaptLunk** [comment]
I seem to have the same kind of issue. Running vscode 1.25.1 on windows 10. Let me know if I can assist with logs or something.

**Comment by ddobson** [comment]
I've also lost the track info and controls that were in the status bar. Running v1.25.1 on Mac OS. Command pallet controls are still working. Happy

**Comment by knie20** [comment]
Same for me. 

MacOS 10.13.5, VScode 1.25.1

**Comment by AKPWebDesign** [comment]
Same problem as these other folks, 1.25.1 on macOS. The problem is also present on the insiders build of Code. Seems like vscode-spotify is just unable to put anything into the status bar.

**Comment by ShyykoSerhiy** [comment]
Spotify has deprecated old Spotify Play Button :
https://developer.spotify.com/community/news/2018/07/19/new-updates-to-spotify-play-button/ .
Unfortunately vscode-spotify relies on spotilocal (Basically Spotify desktop was running a web server on 127.0.0.1 that can accept specific commands.)

**Comment by ShyykoSerhiy** [comment]
On MacOs we can 'fix' this by ONLY using applescript bindings for spotify. To do this change
```json
"spotify.useCombinedApproachOnMacOS": {
          "type": "boolean",
          "default": true,
          "description": "Whether to use experimental applescript + http implementation of spotify client to reduce CPU usage on MacOs. MacOs only."
        },
```
to **false** 
It may be also a good idea to change 
```json
"spotify.statusCheckInterval": {
          "type": "number",
          "default": 5000,
          "description": "Interval of spotify status checks. Frequent status checks may result in faster battery drain and high CPU load (especially on MacOs)."
        }
```
to smth bigger.

**Comment by ShyykoSerhiy** [comment]
Meanwhile I'll accelerate development of https://github.com/ShyykoSerhiy/vscode-spotify/issues/27 that will bring more stable? (there is quite annoying request rate limit https://github.com/spotify/web-api/issues/492) cross platform solution with more features but unfortunately only for Spotify Premium users.

**Comment by guerra08** [comment]
I'm having the same issue on Manjaro 17.1.11 and VSCode 1.25.1.

The keyboard shortcuts are working, only the icons are not being displayed in the status bar.

**Comment by GandalfSaxe** [comment]
Same, macOS 10.14 Beta (18A336e) + VSCode Insider 1.26.

**Comment by ShyykoSerhiy** [comment]
2.5.1 version with useCombinedApproachOnMacOS:false is published. This version fixes the issue for MacOs. Windows/Linux needs #27 to work again.

**Comment by GandalfSaxe** [comment]
Can confirm it works for me again (on macOS).

**Comment by OGNova** [comment]
It seems that the status bar is broken on Windows 10, not sure if it was ever fixed on there, I see fixes for Mac and Linux, but nothing for Windows.

**Comment by nicotheprogrammer** [comment]
For me vscode-spotify does not work. I tried to restart spotify and vscode and reinstalled the extension.

- Windows 10
- vscode version: 1.26.0
- Spotify version: 1.0.87.491.ge2a121fc

**Comment by DanielRios549** [comment]
The extension does not work for me as well, I use Linux Mint 19 Cinnamon, VS Code 1.26.1 and Spotify 1.0.80.480.g51b03ac3

**Comment by ShyykoSerhiy** [comment]
Version 2.6.0 bring back support to Linux via full dbus solution.
Windows version is still on track and is expected to land sometime at the end of this week in 3.0.0-beta version. It will bring support to all platforms via spotify web api. This means couple of things:

- On Windows this extension will only work if you have Spotify Premium (blame Spotify for this. See  https://developer.spotify.com/documentation/web-api/guides/using-connect-web-api/) and only if you are connected to the internet. 
- On other platforms (OSX, Linux) you will be able to switch between local(no internet (applescript, dbus)) implementation and web api implementation.
- Web api implementation will add some new features like : "Add song to favorites", "Add song to playlist".

**Comment by icemouton** [comment]
@ShyykoSerhiy I get what you're saying about dbus in 2.6.0, but I'm still getting : 

> vscode-spotify: This functionality is not supported on this platform.

Using ubuntu 18.04 and latest vscode/vscode-spotify
I have no problem waiting on 3.0 and the new web api plugin, but I figured you would like to know it might not work exactly as you think.

**Comment by ShyykoSerhiy** [comment]
@icemouton when exactly do you get this?

**Comment by DanielRios549** [comment]
I get this problem when I click on play/pause button, but the rest is working perfectly

**Comment by icemouton** [comment]
same as @DanielRios549 with the volme not working but not giving out an error, but play/pause/stop doesn't work, next and previous work fine

**Comment by GandalfSaxe** [comment]
Status bar still works for macOS, but not Windows 10.

**Comment by ShyykoSerhiy** [comment]
@icemouton, @DanielRios549 linux integration should be fixed in 2.6.1. Try in out.

**Comment by icemouton** [comment]
Confirmed working 100% on Ubuntu 18.04. Thanks for the quick fix!

**Comment by DanielRios549** [comment]
Everything is ok on Linux now, only repeating and shuffling do not work (it is on readme). Thanks for fixing this.

**Comment by ShyykoSerhiy** [comment]
@DanielRios549 
Properties like 
[PROPERTY]  org.mpris.MediaPlayer2.Player.readwrite( s:LoopStatus )
[PROPERTY]  org.mpris.MediaPlayer2.Player.readwrite( b:Shuffle )
do exist in org.mpris.MediaPlayer2.spotify. Unfortunately they don't work. And given that according to  [Spotify](https://www.spotify.com/us/download/linux/) 

> Spotify for Linux is a labor of love from our engineers that wanted to listen to Spotify on their Linux development machines. They work on it in their spare time and it is currently not a platform that we actively support.

it's not going to start working anytime soon.

**Comment by DanielRios549** [comment]
@ShyykoSerhiy Understood, the only way is to wait for a update of Spotify, some options do not work or do not exist on Spotify Linux (did I hear tray icon?), but the extension is almost perfect as it is now, not more bugs and the most important options are working fine.

**Comment by ShyykoSerhiy** [comment]
Web Api Support is here. You can check it out here 
https://github.com/ShyykoSerhiy/vscode-spotify/releases/tag/3.0.0-beta-2 .

**Comment by GandalfSaxe** [comment]
I'm not at a windows machine again for some time now, sorry.

**Comment by CaptLunk** [comment]
@ShyykoSerhiy - Works great! Been running smoothly for an hour now. Will try it more tomorrow at work. 
Thanks for your work! 👍

**Comment by OGNova** [comment]
How would one install that release?

**Comment by CaptLunk** [comment]
@OGNova - This guide worked for me: https://github.com/OmniSharp/omnisharp-vscode/wiki/Installing-Beta-Releases

**Comment by OGNova** [comment]
It's working great, the only thing I've found is just a visual thing at most really, https://cdn.aetherya.stream/uploads/k3BeE.png

Not sure what that's supposed to be, but figured I'd share it.

**Comment by ShyykoSerhiy** [comment]
@OGNova, that's quite odd. What do you have in spotify.trackInfoFormat setting? And can you please give a link to this song?

**Comment by OGNova** [comment]
I have `artistName - trackName` for trackInfoFormat, and a link to the song is https://open.spotify.com/track/53AuxddsgPSlqM619nJhSd?si=LWXcCF5DReOwc1SjKf1zAw

**Comment by ShyykoSerhiy** [comment]
#42 issue is fixed in 3.0.0
